modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/OAuth2RestClient.java

```diff
@@ -0,0 +1,163 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.identity.integration.test.restclients;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.restassured.http.ContentType;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.Header;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.message.BasicHeader;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONException;
+import org.testng.Assert;
+import org.wso2.carbon.automation.engine.context.beans.Tenant;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import org.wso2.identity.integration.test.rest.api.server.application.management.v1.model.ApplicationListResponse;
+import org.wso2.identity.integration.test.rest.api.server.application.management.v1.model.ApplicationModel;
+import org.wso2.identity.integration.test.rest.api.server.application.management.v1.model.ApplicationPatchModel;
+import org.wso2.identity.integration.test.rest.api.server.application.management.v1.model.ApplicationResponseModel;
+import org.wso2.identity.integration.test.rest.api.server.application.management.v1.model.OpenIDConnectConfiguration;
+import org.wso2.identity.integration.test.rest.api.server.application.management.v1.model.SAML2ServiceProvider;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class OAuth2RestClient extends RestBaseClient {
+
+    private static final String API_SERVER_BASE_PATH = "api/server/v1";
+    private static final String APPLICATION_MANAGEMENT_PATH = "/applications";
+    private static final String INBOUND_PROTOCOLS_BASE_PATH = "/inbound-protocols";
+    private final String applicationManagementApiBasePath;
+    private final String username;
+    private final String password;
+
+    public OAuth2RestClient(String backendUrl, Tenant tenantInfo) {
+        this.username = tenantInfo.getContextUser().getUserName();
+        this.password = tenantInfo.getContextUser().getPassword();
+
+        String tenantDomain = tenantInfo.getContextUser().getUserDomain();
+        applicationManagementApiBasePath = getApplicationsPath(backendUrl, tenantDomain);
+    }
+
+    public String createApplication(ApplicationModel application) throws IOException, JSONException {
+        String jsonRequest = toJSONString(application);
+
+        try (CloseableHttpResponse response = getResponseOfHttpPost(applicationManagementApiBasePath, jsonRequest,
+                getHeaders())) {
+            String[] self = response.getHeaders(LOCATION_HEADER)[0].toString().split(PATH_SEPARATOR);
```

Comment:
Shall we update the variable name to reflect that this is the resource location?

Reply:
fixed with https://github.com/wso2/product-is/pull/15906/commits/000f146a4083f6818537ad882273d3b633fb5100